src/js/providers/html5.js

```diff
@@ -17,16 +20,37 @@ define([
         _isIOS7 = utils.isIOS(7);
 
 
+
+    // Browsers, including latest chrome, do not always report Stalled events in a timely fashion
+    var stallCheckGenerator = function(videotag, stalledHandler) {
+        var lastChecked = 0;
+        return function() {
+            if (videotag.paused) { return; }
+            var t = videotag.currentTime * 1000; // secs to ms
+            if (t - lastChecked < STALL_TOLERANCE) {
+                stalledHandler();
+            }
+            lastChecked = t;
+        };
+    };
+
     function _setupListeners(eventsHash, videoTag) {
         utils.foreach(eventsHash, function(evt, evtCallback) {
             videoTag.addEventListener(evt, evtCallback, false);
         });
+
+        var checker = stallCheckGenerator(videoTag, eventsHash.stalled);
+        stallInterval = setInterval(checker, STALL_DELAY);
     }
 
     function _removeListeners(eventsHash, videoTag) {
         utils.foreach(eventsHash, function(evt, evtCallback) {
             videoTag.removeEventListener(evt, evtCallback, false);
         });
+
+        if (stallInterval) {
```

Comment:
don't need the if. assign a default value of 0 or -1 (pick your fave) to intervalIds so js knows it's a Number